bundle-core/src/main/java/net/discdd/client/bundlesecurity/ClientSecurity.java

```diff
@@ -230,90 +226,33 @@ public static synchronized ClientSecurity getInstance() throws IllegalStateExcep
     }
 
     /* Encrypts File and creates signature for plain text */
-    public Path[] encrypt(final Path toBeEncPath, final Path encPath, final String bundleID) throws IOException,
-            InvalidKeyException {
-        final var bundlePath = encPath.resolve(bundleID);
-        final var payloadPath = bundlePath.resolve(SecurityUtils.PAYLOAD_DIR);
-        final var signPath = bundlePath.resolve(SecurityUtils.SIGNATURE_DIR);
-        List<Path> returnPaths = new ArrayList<>();
-
-        /* Create Directory if it does not exist */
-        bundlePath.toFile().mkdirs();
-        payloadPath.toFile().mkdirs();
-        signPath.toFile().mkdirs();
-
-        try (DataInputStream inputStream = new DataInputStream(new FileInputStream(toBeEncPath.toFile()))) {
-            byte[] chunk = new byte[SecurityUtils.CHUNKSIZE];
-            int len;
-            for (int i = 1; (len = inputStream.read(chunk)) != -1; i++) {
-                var payloadFilePath = payloadPath.resolve(SecurityUtils.PAYLOAD_FILENAME + i);
-                var signFilePath =
-                        signPath.resolve(SecurityUtils.PAYLOAD_FILENAME + i + SecurityUtils.SIGNATURE_FILENAME);
-
-                /* if we got a partial chunk make a new chunk with the exact size */
-                if (chunk.length != len) chunk = Arrays.copyOf(chunk, len);
-
-                /* Create Signature with plaintext*/
-                createSignature(chunk, signFilePath);
-                /* Encrypt File */
-                CiphertextMessage cipherText = cipherSession.encrypt(chunk);
-                updateSessionRecord();
-                Files.write(payloadFilePath, cipherText.serialize());
-            }
-        }
-
-        /* Create Bundle ID File */
-        var bundleIDPath = bundlePath.resolve(SecurityUtils.BUNDLEID_FILENAME);
-        Files.write(bundleIDPath, bundleID.getBytes());
-
-        /* Write Keys to Bundle directory */
-        Path[] identityKeyPaths = writeKeysToFiles(bundlePath, false);
-
-        returnPaths.add(payloadPath);
-        returnPaths.add(signPath);
-
-        returnPaths.addAll(Arrays.asList(identityKeyPaths));
-        return returnPaths.toArray(new Path[returnPaths.size()]);
+    public CiphertextMessage encrypt(byte[] bytes) {
+        /* Encrypt File */
+        CiphertextMessage cipherText = cipherSession.encrypt(bytes);
+        System.out.println("BRRRRRRRRRR " + ((SignalMessage)cipherText).getCounter());
+        updateSessionRecord();
+        return cipherText;
     }
 
     public void decrypt(Path bundlePath, Path decryptedPath) throws IOException, InvalidMessageException,
```

Comment:
the simplification is great but I dont get the payloads loop part, what was the purpose of it and why we have it simplified?

Reply:
before we were chunking the payload into 1M chunks. now we only have one payload. we still have the chunking problem, but we need to fix that by changing signals encrypt/decrypt to take Input/Outputstreams rather than multiple payloads.